src/parser/ast/mod.rs

```diff
@@ -205,7 +207,7 @@ pub enum Stmt {
         if_not_exists: bool,
         tbl_name: QualifiedName,
         module_name: Name,
-        args: Option<String>, // TODO smol str
+        args: Option<Vec<ModuleArgument>>, // TODO smol str
```

Comment:
@gwenn, you would just join multi-token args with whitespaces?
for example, how would you handle:
```
create virtual table geo using rtree(id integer, x, y);
```

Reply:
Anything except `,` is meaningless. So:
```
['id integer', 'x', 'y']
```
is expected.

See https://sqlite.org/lang_createvtab.html
> he module takes zero or more comma-separated arguments. The arguments can be just about any text as long as it has balanced parentheses. The argument syntax is sufficiently general that the arguments can be made to appear as [column definitions](https://sqlite.org/lang_createtable.html#tablecoldef) in a traditional [CREATE TABLE](https://sqlite.org/lang_createtable.html) statement. SQLite passes the module arguments directly to the [xCreate](https://sqlite.org/vtab.html#xcreate) and [xConnect](https://sqlite.org/vtab.html#xconnect) methods of the module implementation without any interpretation. It is the responsibility of the module implementation to parse and interpret its own arguments.

Reply:
do you have any idea how to actually pass the content of the arguments verbatim? sqlite is returning a reference to the original string, but `yy0` on our side is returning a `String`.  Would joining the args with whitespace be enough? Do you have an idea how to do it properly otherwise?

Reply:
Something like:
- each time [vtab_arg_init](https://github.com/gwenn/lemon-rs/blob/12e846883ab041486b6bb76b1da20e5a11432f56/src/parser/mod.rs#L90) is called, the `Context` must be switch to "accumulating" mode.
- in [Parser::next](https://github.com/gwenn/lemon-rs/blob/12e846883ab041486b6bb76b1da20e5a11432f56/src/lexer/sql/mod.rs#L167), when the "accumulating" mode is active, the token value should be the accumulated one
- stop accumulating when [here](https://github.com/gwenn/lemon-rs/blob/12e846883ab041486b6bb76b1da20e5a11432f56/src/parser/parse.y#L1326)

I will give it a try this week-end.

Reply:
Could you please give #32 a try ?
Tested with:
```diff
diff --git a/examples/sql_cmd.rs b/examples/sql_cmd.rs
index 11b6b39..eb00d1f 100644
--- a/examples/sql_cmd.rs
+++ b/examples/sql_cmd.rs
@@ -5,7 +5,10 @@ use sqlite3_parser::lexer::sql::Parser;
 // RUST_LOG=sqlite3Parser=debug
 fn main() {
     env_logger::init();
-    let arg = "PRAGMA parser_trace=ON;";
+    let arg = r#"CREATE VIRTUAL TABLE mail USING fts3(
+  subject VARCHAR(256) NOT NULL,
+  body TEXT CHECK(length(body)<10240)
+);"#;
     let mut parser = Parser::new(arg.as_bytes());
     loop {
         match parser.next() {
``